src/Packages/Microsoft.Internal.Extensions.DotNetApiDocs.Transport/Microsoft.Internal.Extensions.DotNetApiDocs.Transport.proj

```diff
@@ -0,0 +1,41 @@
+<Project Sdk="Microsoft.Build.NoTargets">
+
+    <PropertyGroup>
+        <TargetFramework>$(LatestTargetFramework)</TargetFramework>
```

Comment:
LatestTargetFramework seems to be the value to use here. In runtime, we also add the target OS at the end, but it does not seem to apply here.

Reply:
We multi-target (currently: net462,netcoreapp3.1,net8.0), so we don't use `TargetFramework` and use `TargetFrameworks` instead.

`LatestTargetFramework` points currently to the `net8.0`, soon it'll be pointing to net9.0. Though, the projects themselves will likely retain 'net462,netcoreapp3.1,net8.0', and only projects that require .NET 9-specific API will be targeting `LatestTargetFramework` (or `net9.0` explicitly). This is something we haven't thought through yet, but we'll have to do so very soon :) /cc: @joperezr

Reply:
You shouldn't set both `TargetFramework` and `TargetFrameworks`.  Since this project is using P2P references to gather output from other projects you should make a choice that's based on what you want to include.

If you want to include 1 set of files for single framework, then set TargetFramework.

If you want to include multiple sets of files for all frameworks, then set TargetFrameworks and make sure it's a superset of all the frameworks targeted by projects referenced.

Reply:
We just need the LatestTargetFramework (= net8.0) for the docs transport package.

Reply:
Maybe not now, but in the future I do envision having APIs which are not present in latest target framework but are present in down level TFMs, so ideally the transport package would have docs per TFM. That is kind of why I asked the overall question before of why can't we just use the process that the docs team already supports where you give them a NuGet package per library, and they are able to extract the docs per TFM for each of them.

Reply:
> why can't we just use the process that the docs team already supports where you give them a NuGet package per library, and they are able to extract the docs per TFM for each of them.

Please elaborate, I'm not sure which process you are referring to. The current model, as @carlossanlop described below, is manual and AFAIK only considers the latest TFM. There is nothing that prevents the transport package from including other TFMs. We could even do that as part of this PR today but the docs team would need to agree on indexing more than just the current TFM.

In general, the docs team wants a single transport package per repository, including the reference assemblies and the compiler generated XML files. We aim towards creating these packages before .NET 8 RC2. In the future, we could/should consider adding centralized infrastructure (presumably in Arcade) to automatically create transport packages based on repository information.

Reply:
Sure. So after chatting with @gewarren a few months ago, we were instructed that the process for pushing api Reference docs to the docs.microsoft.com, we would simply need to create a new workitem here: https://aka.ms/publish-on-docs/reference outlining the packages that need to be updated, versions and the rest of the info that is asked in that template. From there, the Docs team would restore those packages locally, extract the xmls from each of our packages, and then generate the content in the format then need it in in order to publish it to the docs site. We were also told that we would have to repeat this process each time we wanted to push updates for the docs whenever there was a new version of the packages.

All of this process is documented [here](https://review.learn.microsoft.com/en-us/help/onboard/dotnet-requirements?branch=main&branchFallbackFrom=master). The nice thing about this process is that it doesn't really require for us to have to generate anything special, and instead we could just share the NuGet packages and that would be sufficient on their side. Worth noting that this is also the process that we have been following from the dotnet/iot repository for the past few years and it has worked without any issues. Again, I understand why dotnet/runtime's process is more convoluted and hard, and why it requires a special transport package to be created, but what I don't fully understand is why is this required for the extensions repo which is much simpler and basically only ships individual nuget packages.

Reply:
How does this correlate with the api-catalog-infra that today needs to index the extensions packages by manually specifying them here? https://github.com/dotnet/api-catalog-infra/blob/20eadd453545ab028052662436288f4bd2a20f3c/eng/droplayout.props#L149-L172

@carlossanlop

Reply:
I'm not 100% sure (@gewarren or @RussKie  can probably guide us here) but I believe that is a separate step, so in general you submit a workitem for the docs team specifying the packages, and then if the package is new you just add a line there so that the package is also included.

Reply:
I didn't know you had to add an entry to api-catalog-infra if you're generating docs via https://aka.ms/publish-on-docs/reference requests. But anything that ends up in the "extensions" drop when api-catalog-infra is run will be ingested for API doc generation to the https://github.com/dotnet/dotnet-api-docs repo as part of each .NET release.

Reply:
I see, so I guess the uber question is: Is there benefit to having these transport packages being produced from the extensions repo too, or would it be the same thing if we just continue to push our NuGet packages through aka.ms/publish-on-docs/reference requests? @gewarren can you help identify the pros and cons so we can decide what the best approach is for dotnet-extensions?

Reply:
I think the reason we went with api-catalog-infra is because we wanted the docs in the same repo as the other .NET extensions. There's an issue if you have them in a different repo and there are overlapping namespaces (although not sure if that's the case). Then there's also monikers to consider. @v-alje knows more about the pros/cons than me.

Reply:
@gewarren @joperezr Looking at the code in the linked pull request <https://github.com/dotnet/runtime/pull/89312>, it looks like these extensions share namespaces with .NET API reference content currently in the <https://github.com/dotnet/dotnet-api-docs> repo. You can look that up by entering the namespace in the Search box on this page: <https://learn.microsoft.com/en-us/dotnet/api/?branch=main>.  

Libraries and packages which share a namespace need to be processed in the same repo and under the same CI job otherwise one namespace page will take precedence over all the others and the types from the latter will not get linked to in the namespace page.  

So, these extensions need to be processed in the <https://github.com/dotnet/dotnet-api-docs> repo.

Reply:
Gotcha, thanks @v-alje  so much for the explanation and apologies for the questioning, just wanted to make sure that the changes in this PR were indeed required which now I understand they are.